src/loops.rs

```diff
@@ -499,6 +519,79 @@ fn check_for_loop_explicit_counter(cx: &LateContext, arg: &Expr, body: &Expr, ex
     }
 }
 
+// Check for the FOR_KV_MAP lint.
+fn check_for_loop_over_map_kv(cx: &LateContext, pat: &Pat, arg: &Expr, body: &Expr, expr: &Expr) {
+    if let PatTup(ref pat) = pat.node {
+        if pat.len() == 2 {
+
+            let (pat_span, kind) = match (&pat[0].node, &pat[1].node) {
+                (key, _) if pat_is_wild(key, body) => (&pat[1].span, "values"),
+                (_, value) if pat_is_wild(value, body) => (&pat[0].span, "keys"),
+                _ => return
+            };
+
+            let ty = walk_ptrs_ty(cx.tcx.expr_ty(arg));
+            let arg_span = if let ExprAddrOf(_, ref expr) = arg.node {
+                expr.span
+            }
+            else {
+                arg.span
+            };
+
+            if match_type(cx, ty, &HASHMAP_PATH) ||
+               match_type(cx, ty, &BTREEMAP_PATH) {
+                span_lint_and_then(cx,
+                          FOR_KV_MAP,
+                          expr.span,
+                          &format!("you seem to want to iterate on a map's {}", kind),
+                          |db| {
+                    db.span_suggestion(expr.span,
+                                       "use the corresponding method",
+                                       format!("for {} in {}.{}()",
```

Comment:
Since this is a suggestion and will show surrounding code as well, perhaps it should be ended with a `{...}` (technically, `{{...}}` to escape)? Otherwise LGTM